src/api/app/helpers/webui/request_helper.rb

```diff
@@ -174,4 +174,23 @@ def request_type_of_action(bs_request)
 
     bs_request.bs_request_actions.first.type.titleize
   end
+
+  def quick_build_results_status(buildresults)
+    return unless buildresults.reject { |b| b == 'succeeded' }.any?
```

Comment:
Wouldn't it be faster computation with
`return unless buildresults.any? { |b| b != 'succeeded' }` ?

Reply:
Definitely!